configs/module-defaults.config

```diff
@@ -13,12 +13,17 @@ params {
             publish_results  = "all"
         }
         'guppy_basecaller' {
-            args             = "--compress_fastq"
+						args = "--compress_fastq --save_path . --records_per_fastq 0"
```

Comment:
can we keep `--save_path .` in the main module?  It has to be this setting to work with nextflow writing to the work directory

Reply:
I snuck the `--save_path .` option into the config file, since that will be constant between the CPU and GPU versions. I passed a few arguments there that will be the same between the two versions (`--compress_fastq` and `--records_per_fastq 0`). This may be at odds with nextflow design standards though.

Reply:
The structure of the guppy .cfg file looks like so (this is what guppy calls automatically when you invoke it):
`
# Basic configuration file for ONT Guppy basecaller software.

# Compatibility
compatible_flowcells                = FLO-FLG001,FLO-MIN106
compatible_kits                     = SQK-CAS109,SQK-CS9109,SQK-DCS108,SQK-DCS109,SQK-LRK001,SQK-LSK108,SQK-LSK109,SQK-LSK109-XL,SQK-LWP001,SQK-PCS108,SQK-PCS109,SQK-PRC109,SQK-PSK004,SQK-RAD002,SQK-RAD003,SQK-RAD004,SQK-RAS201,SQK-RLI001,VSK-VBK001,VSK-VSK001,VSK-VSK002
compatible_kits_with_barcoding      = SQK-16S024,SQK-PCB109,SQK-RBK001,SQK-RBK004,SQK-RLB001,SQK-LWB001,SQK-PBK004,SQK-RAB201,SQK-RAB204,SQK-RPB004,VSK-VMK001,VSK-VMK002

# Data trimming.
trim_strategy                       = dna
trim_threshold                      = 2.5
trim_min_events                     = 3

# Basecalling.
model_file                          = **template_r9.4.1_450bps_hac.jsn**
chunk_size                          = 2000
gpu_runners_per_device              = 4
chunks_per_runner                   = 512
chunks_per_caller                   = 10000
overlap                             = 50
qscore_offset                       = 0.421
qscore_scale                        = 0.879
builtin_scripts                     = 1

# Calibration strand detection
calib_reference                     = lambda_3.6kb.fasta
calib_min_sequence_length           = 3000
calib_max_sequence_length           = 3800
calib_min_coverage                  = 0.6

# Output.
records_per_fastq                   = 4000
min_qscore                          = 7.0

# Telemetry
ping_url                            = https://ping.oxfordnanoportal.com/basecall
ping_segment_duration               = 60

# Adapter scaling
as_model_file			    = adapter_scaling_dna_r9.4.1_min.jsn

`

But, if you overwrite any of these options by passing them as arguments on the command line, those will take precedence. I guess there are two options: we can either pass command line arguments using the nextflow config file, or we can pass fewer arguments within nextflow and instead bundle it with several guppy .cfg files. II suppose the perfect scenario would be to design custom guppy configs with reasonable defaults for the specific GPU they're running on (since the total amount of VRAM varies between GPUs); but it is difficult to predict _a priori_ how much memory a nanopore run will take up, and if it exceeds the total device memory, guppy will crash. I defer to your expertise here as to which option is preferable.

Note to self: May run into issues with paths when using custom config files, as the corresponding model file (bolded) might need to also be copied out of the default `/opt/ont/guppy/data/` directory.

Reply:
yeah i think lets go with the nextflow config files - its confusing having multiple configs i reckon. the compress fastq and records per fastq are suggested in the manual so i put them as defaults. The reason `--save_path` is hard coded is because it can't change - if the user changes it in the module defaults, the program will fail so its got to be in both processes. 

We could include functionality to provide as custom cfg file though a channel?

Reply:
i think for now though, lets just have these defaults in a different config file.

Reply:
Could you do a test of loading the default-modules.config and then loading a custom.config on the next line and check that only the guppy module params are overwritten? my fear is that the whole module param block will be overwritten

Reply:
I have moved `--save_path .` back to the modules.

I like the idea of including a custom guppy config channel, although I'd say it's a low priority for now (we usually just use the guppy defaults - I only changed them so that I could get things running on my workstation).

I am not entirely sure what you meant by loading a custom config. If I separate out the nextflow-format config file for t the guppy module, and import this config file after luslab-base, it respsects the parameters. Please see the `oist-assembler`:

[oist-assembler](https://github.com/luslab/oist-assembler)

Did you mean that luslab-modules/tools/guppy should include separate nextflow-format custom.config files?




As an aside, in my last commit, I have subset the lambda test fast5 dataset to include only the 10 longest reads. This is sufficient for Flye to assemble it and simplifies computation. I also added some logging to the Flye module. But that is a different - maybe this pull request should be closed.